reconcile/gitlab_housekeeping.py

```diff
@@ -83,88 +83,90 @@ def handle_stale_items(dry_run, gl, days_interval, enable_closing, item_type):
                     gl.remove_label(item, item_type, LABEL)
 
 
-def is_good_to_merge(labels):
-    return any(l in MERGE_LABELS for l in labels) and \
+def is_good_to_merge(merge_label, labels):
+    return merge_label in labels and \
         not any(l in HOLD_LABELS for l in labels)
 
 
 def rebase_merge_requests(dry_run, gl, rebase_limit):
     mrs = gl.get_merge_requests(state='opened')
```

Comment:
Does it make sense to move the `get_merge_requests` inside the `for merge_label in MERGE_LABELS_PRIORITY` and [filter per label](https://docs.gitlab.com/ee/api/merge_requests.html#list-merge-requests)? That would avoid processing MRs without the current label in the loop, but would also generate additional calls to the API to list MRs. Not sure it pays off.

Reply:
right now it's only doing internal processing inside the loop, so i think we can keep things as they are to save some API calls

Reply:
There's the `gl.project.commits.list`